tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -2184,8 +2168,19 @@ SpirvInstruction *SpirvEmitter::processCall(const CallExpr *callExpr) {
       // do not need to mark the "param.var.*" variables as precise.
       const bool isPrecise = false;
 
-      auto *tempVar =
-          spvBuilder.addFnVar(varType, arg->getLocStart(), varName, isPrecise);
+      SpirvVariable *tempVar = nullptr;
+      if (isOpaqueArrayType(param->getType()) &&
+          !param->getType()->isConstantArrayType()) {
+        // If it is a bindless array of an opaque type, we have to use
+        // a pointer to a pointer of the runtime array.
+        tempVar = spvBuilder.addFnVar(
+            spvContext.getPointerType(varType,
+                                      spv::StorageClass::UniformConstant),
+            arg->getLocStart(), varName, isPrecise);
+      } else {
+        tempVar = spvBuilder.addFnVar(varType, arg->getLocStart(), varName,
+                                      isPrecise);
+      }
```

Comment:
I think the above logic should be performed inside `addFnVar`. That way, anyone calling `addFnVar` will get the right behavior for free. Same as above. What do you think?

Reply:
I agree. Done.